semicolon in comma-first setup regression from 0.24.1

I got following issue (in 1.1.0, worked perfectly in 0.24.1):

eslint config:

```
    "indent": [2, 2]
```

code:

```
11 var path     = require('path')
12  , crypto    = require('crypto')
13  ;
```

output:

```
   13:4  error  Expected indentation of 0 characters but found 2  indent
```

Thank you.

PS. Discussed in greater details here: https://github.com/eslint/eslint/issues/3391

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -472,6 +472,7 @@ module.exports = function(context) {
     function checkIndentInVariableDeclarations(node) {
         var elements = filterOutSameLineVars(node);
         var nodeIndent = getNodeIndent(node);
+        var lastElement = elements[elements.length - 1];
 
         var elementsIndent = nodeIndent + indentSize * options.VariableDeclarator[node.kind];
 
@@ -480,12 +481,18 @@ module.exports = function(context) {
 
         if (elements.length > 0) {
             // Skip last block line check if last item in same line
-            if (elements[elements.length - 1].loc.end.line === node.loc.end.line) {
+            if (lastElement.loc.end.line === node.loc.end.line) {
                 return;
             }
         }
 
-        checkLastNodeLineIndent(node, elementsIndent - indentSize);
+        var tokenBeforeLastElement = context.getTokenBefore(lastElement);
+        if (tokenBeforeLastElement.value === ",") {
+            // Special case for comma-first syntax where the semicolon is indented
+            checkLastNodeLineIndent(node, getNodeIndent(tokenBeforeLastElement));
+        } else {
+            checkLastNodeLineIndent(node, elementsIndent - indentSize);
+        }
     }
 
     /**

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -682,6 +682,19 @@ ruleTester.run("indent", rule, {
                 "  }\n" +
                 "};",
             options: [2]
+        },
+        {
+            code:
+                "var path     = require('path')\n" +
+                "  , crypto    = require('crypto')\n" +
+                "  ;\n",
+            options: [2]
+        },
+        {
+            code:
+                "var a = 1\n" +
+                "   ,b = 2\n" +
+                "   ;"
         }
     ],
     invalid: [
@@ -1236,6 +1249,25 @@ ruleTester.run("indent", rule, {
                 [3, 6, 7, "Property"],
                 [4, 6, 8, "Property"],
                 [5, 4, 5, "ObjectExpression"]
+            ])
+        },
+        {
+            code:
+                "var path     = require('path')\n" +
+                " , crypto    = require('crypto')\n" +
+                ";\n",
+            options: [2],
+            errors: expectedErrors([
+                [3, 1, 0, "VariableDeclaration"]
+            ])
+        },
+        {
+            code:
+                "var a = 1\n" +
+                "   ,b = 2\n" +
+                ";",
+            errors: expectedErrors([
+                [3, 3, 0, "VariableDeclaration"]
             ])
         }
     ]

